074-connector-offsets-support.md

```diff
@@ -0,0 +1,226 @@
+
+For a sink connector it might look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-sink-connector-offsets
+data:
+  offsets.json: |
+    {
+      "offsets": [
+        {
+          "partition": {
+            "kafka_topic": "my-topic"
+              "kafka_partition": 2
+          },
+          "offset": {
+            "kafka_offset": 4
+          }
+        }
+      ]
+    }
+```
+
+### Altering offsets
+
+To alter offsets the user will annotate the KafkaConnector resource with the new annotation `strimzi.io/connectorOffsets` set to `alter`.
+After the annotation is added, on the next reconciliation the operator will call the PATCH /connectors/{connector}/offsets endpoint to alter the offsets.
+Once the patch is complete the operator will then remove the annotation from the KafkaConnector CR.
+
+By default, the operator will construct the request body from a ConfigMap called `{connector}-offsets`, where `{connector}` is the name of the KafkaConnector CR that was annotated.
+This matches the name used in the list operation, so that a user can list offsets, then edit the ConfigMap the operator created, then request the offsets be altered using that same ConfigMap.
+
+The KafkaConnector CRD will also be updated to contain a new property called `alterOffsetsCM` that when set will override the name of the ConfigMap the operator uses to populate the request body.
+
+Notes:
+* The format of the request body will be validated to make sure it is valid JSON, but the operator won't perform any further validation, instead relying on the API endpoint to return a reasonable error message.
+* If the request fails the KafkaConnector CR status will be updated to include a warning message with the response from the API, e.g.
+  ```yaml
+  - lastTransitionTime: "2024-06-04T08:44:15.913138115Z"
+    message: Failed to alter the connector offsets due to "message from endpoint".
+    reason: AlterOffsets
+    status: "True"
+    type: Warning
+  ```
+* Strimzi will not attempt to verify that the connector is already stopped, it will rely on the API failing if that is the case.
+  This is to simplify the logic and means we don't have to account for what happens if the KafkaConnector spec has been updated but the connector isn't stopped yet.
+
+### Resetting offsets
+
+To reset the offsets for a particular connector the user will annotate the KafkaConnector resource with the new annotation `strimzi.io/connectorOffsets` set to `reset`.
+When the operator observes this annotation it will use the DELETE /connectors/{connector}/offsets endpoint to reset all offsets for the connector.
+When the reset has been completed the operator will remove this annotation from the resource.
+
+If the reset fails the operator will add a condition to the KafkaConnector resource called `ResetOffsets` to communicate the failure and remove the annotation.
+
+Strimzi will shortcut and automatically fail to do the reset if the KafkaConnector resource is not stopped.
```

Comment:
Good spot, that is because previously the alter was being done in a different resource, not in KafkaConnector CR. We can check ourselves in both cases now, since the state should be correct in the KafkaConnector, so I'll update it to say that in both cases.